### ISSUE TITLE
Investigate potential bug with progress/completion

### ISSUE THREAD

**Issue by marcellamaki** [learningequality/kolibri#8309]
There may be a bug in develop with progress/completion. It seems like there are mismatches between the progress that is displayed in the UI and the progress that is calculated, and that some resources cannot be successfully "completed". So far, it seems to be occurring in the epub content renderer, and possibly with videos. 

The purpose of this issue is to investigate and see if the issue is reproducible, and investigate where the problem might be happening, to create a detailed follow up issue.

![image](https://user-images.githubusercontent.com/17235236/129750443-5efe38a1-a460-4add-afda-91c57eedadf7.png)

**Comment by rtibbles** [comment]
For additional context here, I can imagine that EPUBs specifically may be more problematic, but there are circumstances under which PDFs may also pose apparent issues to the user.

For both EPUBs and PDFs we implemented an 'all pages visited' criterion. For EPUBs the specific implementation may cause problems because EPUBs are reflowable, and so what constitutes a page can change over time. We may need to revisit how we are tracking this and make sure that it is indeed matching up with the internal EPUBJS mechanics tracking where the user is.

For PDFs, because of the way that we defer rendering of each page until you stop scrolling on that page (to reduce browser load and keep PDF rendering performant even for large PDFs), you will only register as having 'visited' a page if you stop on that page during scrolling and wait for it to load.

These issues will be particularly acute for devs who are just skipping through content and not actually reading it.

Recent work on the video renderer did very little except allow purely 'time based' progress tracking to be forced by an optional flag.


If there is a holistic issue with inconsistent progress tracking across all content types, it is likely a bug with 'forceDurationBasedProgress' being accidentally set to `true`, even when it is undefined, and all progress tracking just being on the basis of having spent X amount of time with it.

**Comment by MisRob** [comment]
I've been encountering the following issue with video completion:

**Steps to reproduce (tested on release-v0.15.x)**
1. I'm signed in and in one of my classes in "Learn" start watching a 60 seconds long video I've never seen before
2. When I've seen the first 30 seconds, I pause the video and rewind it back **_around 20 seconds_** (so I'm at 10 seconds of the video length)
3. I hit the play button
4. When I'm at around 45 seconds of the video (**_around 15 seconds_** before finishing it), the in-progress icon in the top right corner changes to the completed star and the completion modal is displayed, covering the area of the video that is still playing

_It can't be seen on the screenshot but the video is still playing here:_
![completion](https://user-images.githubusercontent.com/13509191/133409266-5816e815-07bd-415b-9413-aae19eb6682d.png)

@rtibbles mentioned that we're using cumulated time as completion criteria for videos. Although the numbers don't fit precisely (maybe caused by slight delays in UI?), similar timing problems are a recurring issue with videos, and the two numbers I highlighted in the steps to reproduce are always very close to each other so I assume the current completion criteria is the cause.

In the past, it probably wasn't so disruptive since we used snackbars to report completion. Now, with the completion modal, it creates a really bad experience.

A simple solution might be maybe to continue using accumulated time but in addition to it, wait for the video to actually finish before changing its status to complete?
